File: uniapp/lib/models/stop_model.dart
class StopModel {
  int? direction;
  String? name;
  String? routeId;
  List<Stops>? stops;

  StopModel({this.direction, this.name, this.routeId, this.stops});

  StopModel.fromJson(Map<String, dynamic> json) {
    direction = json['direction'];
    name = json['name'];
    routeId = json['route_id'];
    if (json['stops'] != null) {
      stops = <Stops>[];
      json['stops'].forEach((v) {
        stops!.add(Stops.fromJson(v));
      });
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['direction'] = direction;
    data['name'] = name;
    data['route_id'] = routeId;
    if (stops != null) {
      data['stops'] = stops!.map((v) => v.toJson()).toList();
    }
    return data;
  }
}

class Stops {
  String? details;
  String? id;
  double? latitude;
  double? longitude;
  String? place;

  Stops({details, id, latitude, longitude, place});

  Stops.fromJson(Map<String, dynamic> json) {
    details = json['details'];
    id = json['id'];
    latitude = json['latitude'];
    longitude = json['longitude'];
    place = json['place'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['details'] = details;
    data['id'] = id;
    data['latitude'] = latitude;
    data['longitude'] = longitude;
    data['place'] = place;
    return data;
  }
}

File: uniapp-api/app/models/StopModel.py
from database.db import get_connection
from .entities.Stop import Stop


class StopModel:
    @classmethod
    def get_stops(cls):
        try:
            connection = get_connection()
            stops = []

            with connection.cursor() as cursor:
                cursor.execute("select rt.id_route, "
                               "rt.route_name, "
                               "rt.direction, "
                               "bs.id_bus_stop, "
                               "bs.place, "
                               "bs.details, "
                               "bs.latitude, "
                               "bs.length "
                               "from bus_stop as bs "
                               "inner join route_bus_stop rbs "
                               "on bs.id_bus_stop = rbs.id_bus_stop "
                               "inner join route as rt "
                               "on rt.id_route = rbs.id_route")
                result = cursor.fetchall()

                for row in result:
                    stop = Stop(row[0], row[1], row[2], row[3], row[4], row[5], row[6], row[7])
                    stops.append(stop.to_json())

            connection.close()
            return stops
        except Exception as ex:
            raise Exception(ex)
